BJS2-80575 Meta-вселенная?

DimonDem123 opened the pull request:
В принципе сделал. Что как работает допер, если как-то можно исправить или сделать лучше попрошу прокоментировать. Но не факт что лапки дойдут, для меня главное что тему понял. Хорошего дня

DimonDem123 committed:
BJS2-80575
В принципе сделал. Что как работает допер, если как-то можно исправить или сделать лучше попрошу прокоментировать. Но не факт что лапки дойдут, для меня главное что тему понял.
Хорошего дня

DimonDem123 committed:
Update Main.java

переделал чтобы прошло по длине строк

DimonDem123 committed:
Update Main.java

кто бы мог подумать забыл ; поставить

DimonDem123 committed:
Update Main.java

iBerbasov commented on src/main/java/school/faang/bjs280575/NotificationManager.java:
```diff
@@ -0,0 +1,58 @@
+
+@Slf4j
+public class NotificationManager {
+    private Map<NotificationType, Consumer<Notification>> mapManager = new HashMap<>();
```
это не менеджер не мапы, а хендлеры
handlersMap, handlers более подходящие имена

DimonDem123 replied:
Sorry, I don't know English.

DimonDem123 replied:
Я его только учу и и пока додумался что из себя она представляет было лень менять название

iBerbasov commented on src/main/java/school/faang/bjs280575/NotificationManager.java:
```diff
@@ -0,0 +1,58 @@
+            return;
+        } else {
+            Predicate<Notification> findFoulLanguage = not -> {
+                String massag = not.getMessage();
```
message, а не massag

iBerbasov commented on src/main/java/school/faang/bjs280575/NotificationManager.java:
```diff
@@ -0,0 +1,58 @@
+                    return true;
+                }
+            };
+            Function<Notification, Notification> addSignatur = notif -> {
```
Тут тоже не хватает букв в словах, к чему эта экономия ? )

DimonDem123 replied:
я учу анкгийский. И не очень понимаю как правильно называются слова на другом языке, а в гуг переводчик заходить долго

iBerbasov replied:
Ты в ИДЕЕ работаешь? Мне ИДЕЯ часто сама подсказывает правильное написание, когда я забываю какие-то слова ) в гугл редко приходится ходить

iBerbasov commented on src/main/java/school/faang/bjs280575/NotificationManager.java:
```diff
@@ -0,0 +1,58 @@
+                    return true;
+                }
+            };
+            Function<Notification, Notification> addSignatur = notif -> {
```
Всю эту функцию можно так же вынести за пределы класса, по аналогии с фильтрами слов и хендлерами

iBerbasov commented on src/main/java/school/faang/bjs280575/NotificationManager.java:
```diff
@@ -0,0 +1,58 @@
+            log.warn("Таких уведоблений нету.");
+            return;
+        } else {
+            Predicate<Notification> findFoulLanguage = not -> {
```
Это можно вынести за пределы класса и добавлять фильтры по аналогии с registerHandler
сделать List<Predicate<Notification>> filters и добавлять извне фильтры. Так класс станет свободно расширяемым и внешние изменения (если например появятся новые фильтры) не будут влиять на класс (вспоминаем SOLID, первую букву)

iBerbasov commented on src/main/java/school/faang/bjs280575/Main.java:
```diff
@@ -0,0 +1,14 @@
+
+public class Main {
+    public static void main(String[] args) {
+        NotificationManager dataManager = new NotificationManager();
```
Класс NotificationManager, а переменная dataManager - такие разношерстные имена вносят много путаницы в коде

DimonDem123 replied:
То есть желательно зазыватья объект класса как он есть, если он один, если их несколько просто придумать соответствующие названия?

iBerbasov commented on src/main/java/school/faang/bjs280575/Main.java:
```diff
@@ -0,0 +1,14 @@
+            System.out.println("Push уведомление: " + notification.getMessage());
+        });
+        dataManager.sendNotification(new Notification(NotificationType.PUSH, "Бака как ты мог купить 4 пельменя"));
+        String message = "Вы звонили нам неделю назад с прозьбой.\n Она еще актуальна?";
```
"с про**С**ьбой", старайся не допускать опечаток и ошибок в коде

DimonDem123 replied:
Ок. Отличное замечание

iBerbasov commented on src/main/java/school/faang/bjs280575/NotificationManager.java:
```diff
@@ -0,0 +1,58 @@
+
+
+        if (mapManager.get(notification.getType()) == null) {
+            log.warn("Таких уведоблений нету.");
```
уведоМлений

DimonDem123 replied:
Моя вина

DimonDem123 committed:
BJS2-80575
Ответил на коменты, переделал под требования. Вывел фильтры и изменение текста сообщения в отдельные методы

DimonDem123 committed:
Update Main.java

Добавил переменную чтобы влезло прошло по длине строки

DimonDem123 committed:
Update NotificationManager.java

DimonDem123 committed:
Update Main.java

DimonDem123 committed:
Update Main.java

iBerbasov commented on src/main/java/school/faang/bjs280575/NotificationManager.java:
```diff
@@ -0,0 +1,66 @@
+        return findFoulLanguage.test(notification);
+    }
+    
+    protected Notification addSignatureTheMessage(Notification notification) {
```
а зачем тут protected, это же чисто приватные методы

iBerbasov commented on src/main/java/school/faang/bjs280575/NotificationManager.java:
```diff
@@ -0,0 +1,66 @@
+@Slf4j
+public class NotificationManager {
+    private Map<NotificationType, Consumer<Notification>> handlersMap = new HashMap<>();
+    public static Set<String> foulLanguage = Set.of(
```
Тут тоже лучше сделать поле приватным

iBerbasov reviewed (approved):
По решению хорошо, в будущих задачах только следи за модификаторами доступа, где достаточно private - лучше его и оставлять. Лишнего наружу не надо отдавать
